### ISSUE TITLE
Please consider packaging Virtualbox host platform

### ISSUE THREAD

**Issue by pceiley** [clearlinux/distribution#390]
**Is your feature request related to a problem? Please describe.**
Sometimes I need to run Windows in a VM

**Describe the solution you'd like**
Virtualbox host packages and relevant kernel modules

**Describe alternatives you've considered**
KVM - but graphics/interaction is still awful with Windows guests.

**Additional context**
https://www.virtualbox.org/wiki/Linux_Downloads

**Comment by ahkok** [comment]
Guest additions are part of the LTS kernel.

**Comment by pceiley** [comment]
Thanks, but I wish to use clear Linux as a VirtualBox host rather than a guest.

**Comment by thiagomacieira** [comment]
Of course, we should have VirtualBox itself too. But the application can be built by the user, whereas kernel modules can't be added without turning secure boot off.

**Comment by miguelinux** [comment]
Have you consider to use the virt-manager from libvirtd.
I know that it is KVM, but I use it with SPICE protocol on Windows 10.

**Comment by pceiley** [comment]
Haven't tried it in a while. What app do you use to interact with the VM? Does gnome boxes work well enough with SPICE these days?

**Comment by bryteise** [comment]
I'll be taking a shot at adding this.

**Comment by pceiley** [comment]
Any luck, @bryteise? Is this likely to be doable?

**Comment by dbdtchkrbrty** [comment]
This is really a necessary addition to the OS itself. They should add this asap.

**Comment by theflakes** [comment]
This is a very important addition for me and the work I do. I'm not a systems developer. I write a lot of code that has to run across Windows and Linux. Therefore the ability to run virtual Linux guests and Windows guest on Clear is necessary for me. Really love the look and feel of CL.

**Comment by jonagoldman** [comment]
Vortual Box is also a must for me as PHP/JS developer together with Vagrant.

**Comment by miguelinux** [comment]
Have you considered virt-manager?

**Comment by jonagoldman** [comment]
@miguelinux no, haven't tried it, Virtual Box has always being part of my development environment and I'm very happy with it, no plans to change.

**Comment by sharp-dev** [comment]
I want to join! Very interested in ClearLinux, but VirtualBox is also needed. Only this hypervisor is free, while at the same time it allows you to easily move virtual machines between hosts running Linux and Windows.

**Comment by JeremyMorgan** [comment]
What can we do to help make this happen? I'd be willing to donate some cycles to making this work. Is there any documentation / info on building Clear packages?

**Comment by bryteise** [comment]
I'll try and give out a dump of where I got to. The documentation for building things in Clear Linux is described [here](https://github.com/clearlinux/distribution/blob/master/contributing.md).

**Comment by bryteise** [comment]
My current work in [progress](https://github.com/bryteise/VirtualBox) (though not much work or progress has gone on for awhile). It seems to be an issue with the kmk binary trying to run pkg-config for cflags and libs. It maybe is trying to use internal versions based on an strace path that don't exist and getting the whole thing terribly wrong.

**Comment by ztheory** [comment]
Bounty Source now up with $100 start: https://www.bountysource.com/issues/69201334-please-consider-packaging-virtualbox-host-platform
